#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: fix for sparc32 compilation
# DP: Author: GOTO Masanori <gotom@debian.org>
# DP: Upstream status: Not in cvs.
# DP: Status Details: waiting for upstream submit.
# DP: Date: 2003-07-08

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- linuxthreads/sysdeps/unix/sysv/linux/sparc/sparc32/sysdep-cancel.h	2003-07-08 23:26:32.000000000 +0900
+++ linuxthreads/sysdeps/unix/sysv/linux/sparc/sparc32/sysdep-cancel.h.new	2003-07-09 22:22:55.000000000 +0900
@@ -68,7 +68,7 @@
 # ifdef IS_IN_libpthread
 #  define CENABLE	call __pthread_enable_asynccancel
 #  define CDISABLE	call __pthread_disable_asynccancel
-# elif defined !NOT_IN_libc
+# elif !defined NOT_IN_libc
 #  define CENABLE	call __libc_enable_asynccancel
 #  define CDISABLE	call __libc_disable_asynccancel
 # else
